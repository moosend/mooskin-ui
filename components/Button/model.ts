import { IButtonBoxComponentProps } from '../Box/model';

export interface IButtonComponentProps extends IButtonBoxComponentProps {
	/** Button href */
	href?: string;

	/** Button size */
	buttonSize?: 'lg' | 'md' | 'sm';
}
