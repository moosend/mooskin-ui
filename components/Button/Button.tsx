import * as React from 'react';

// Mooskin Context HoC that passes context to component props
import { withMooskinContext } from '../Styled/MooskinContextProvider';

// Models
import { IBoxComponentProps } from '../Box/model';
import { IButtonComponentProps } from './model';

// Styled Components
import { StyledButtonIcon, StyledButtonInverse, StyledButtonNormal } from './styles';

/**
 * Button
 */
export const Button: React.FC<IButtonComponentProps> = withMooskinContext((props) => {
	const onClick = (e: React.MouseEvent<HTMLButtonElement>) => {
		!props.disabled && props.onClick && props.onClick(e);
	};
	const ButtonComponent = props.inverseStyle ? StyledButtonInverse : StyledButtonNormal;
	return <ButtonComponent {...props} onClick={onClick} boxAs={props.href ? 'a' : 'button'} />;
});

Button.defaultProps = {
	buttonSize: 'md',
	className: '',
	style: {},
	type: 'button',
};

Button.displayName = 'Button';

/**
 * ButtonIcon
 */
export const ButtonIcon: React.FC<IBoxComponentProps> = (props) => {
	return <StyledButtonIcon {...props} className={`notranslate ${props.className}`} />;
};

ButtonIcon.defaultProps = {
	className: '',
	style: {},
};

ButtonIcon.displayName = 'ButtonIcon';
