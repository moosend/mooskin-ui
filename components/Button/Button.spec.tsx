import toJson from 'enzyme-to-json';
import * as React from 'react';
import Button from './Button';

import { shallow } from 'enzyme';

describe('Button', () => {

    test('renders correctly', () => {
        const func = jest.fn();

        const tree = shallow(
            <Button
                onClick={func}
                disabled
                className="myClass"
                style={{color: 'blue'}}
                inverseStyle
                id={'button1'}
                href={'www.moosend.com'}
                type={'submit'}
            >
                Mooskin
            </Button>
        );
        expect(toJson(tree)).toMatchSnapshot();
    });

    test('renders properly into dom with color and label', () => {
        const func = jest.fn();

        const component = shallow(<Button onClick={func}>asd</Button>);

        expect(component.find('button').text()).toBe('asd');
        expect(component.find('button').prop('disabled')).not.toBe(true);
    });

    test('renders a disabled button if disabled prop is passed', () => {
        const func = jest.fn();

        const component = shallow(<Button  onClick={func} disabled>asd</Button>);

        expect(component.find('[disabled=true]').length).toBe(1);
    });

    test('onClick prop callback is called when clicked', () => {
        const func = jest.fn();

        const component = shallow(<Button  onClick={func}>asd</Button>);
        component.find('button').simulate('click');
        expect(func).toHaveBeenCalled();
    });
});
