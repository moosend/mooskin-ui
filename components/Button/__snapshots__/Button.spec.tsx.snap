// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<Button
  buttonSize="md"
  className="myClass"
  disabled={true}
  href="www.moosend.com"
  id="button1"
  onClick={[MockFunction]}
  style={
    Object {
      "color": "blue",
    }
  }
  type="submit"
>
  <Component
    buttonSize="md"
    className="myClass"
    disabled={true}
    href="www.moosend.com"
    id="button1"
    onClick={[MockFunction]}
    palette={
      Object {
        "backgroundColors": Object {
          "accountNavigation": "#FFFFFF",
          "altWhite": "#FFFFFF",
          "autosExpression": "#ffffff",
          "background": "#FAFAFA",
          "black": "#000000",
          "blue400": "#4299E1",
          "checkboxUnselected": "rgba(0, 0, 0, 0.62)",
          "client": "#E0E0E0",
          "createNew": "#FFFFFF",
          "darkgray1": "#777777",
          "darkgray2": "#454545",
          "darkgray3": "#424242",
          "dropdown": "#FFFFFF",
          "gray1": "#E0E0E0",
          "gray100": "#F0F0F0",
          "gray2": "#E7E7E7",
          "grayAlpha": "rgba(0,0,0,0.17)",
          "grayAlpha100": "rgba(0,0,0,0.06)",
          "grayAlpha400": "rgba(0,0,0,0.36)",
          "green600": "#2F855A",
          "inverseText": "#212121",
          "medgray1": "#757575",
          "medgray2": "#888888",
          "menuDark": "#21262A",
          "menuLight": "#2E3841",
          "orange400": "#ED8936",
          "pink400": "#ED64A6",
          "primary1": "#5548D9",
          "primary2": "#212121",
          "primary3": "#00796B",
          "primary4": "#D32F2F",
          "primary5": "#f0b941",
          "purple400": "#9f7aea",
          "red400": "#f56565",
          "red600": "#C53030",
          "secondary1": "#2E7D32",
          "secondary2": "#757575",
          "secondary3": "#673AB7",
          "secondary4": "#9C27B0",
          "secondary5": "#00796B",
          "secondary6": "#ECF0FC",
          "submenu": "#FFFFFF",
          "teal400": "#38B2AC",
          "teal500": "#02999A",
          "text": "#212121",
          "white": "#FFFFFF",
        },
        "borderColors": Object {
          "accountNavigation": "#FFFFFF",
          "altWhite": "#FFFFFF",
          "autosExpression": "#ffffff",
          "background": "#FAFAFA",
          "black": "#000000",
          "blue400": "#4299E1",
          "checkboxUnselected": "rgba(0, 0, 0, 0.62)",
          "client": "#E0E0E0",
          "createNew": "#FFFFFF",
          "darkgray1": "#777777",
          "darkgray2": "#454545",
          "darkgray3": "#424242",
          "dropdown": "#FFFFFF",
          "gray1": "#E0E0E0",
          "gray100": "#F0F0F0",
          "gray2": "#E7E7E7",
          "grayAlpha": "rgba(0,0,0,0.17)",
          "grayAlpha100": "rgba(0,0,0,0.06)",
          "grayAlpha400": "rgba(0,0,0,0.36)",
          "green600": "#2F855A",
          "inverseText": "#212121",
          "medgray1": "#757575",
          "medgray2": "#888888",
          "menuDark": "#21262A",
          "menuLight": "#2E3841",
          "orange400": "#ED8936",
          "pink400": "#ED64A6",
          "primary1": "#5548D9",
          "primary2": "#212121",
          "primary3": "#00796B",
          "primary4": "#D32F2F",
          "primary5": "#f0b941",
          "purple400": "#9f7aea",
          "red400": "#f56565",
          "red600": "#C53030",
          "secondary1": "#2E7D32",
          "secondary2": "#757575",
          "secondary3": "#673AB7",
          "secondary4": "#9C27B0",
          "secondary5": "#00796B",
          "secondary6": "#ECF0FC",
          "submenu": "#FFFFFF",
          "teal400": "#38B2AC",
          "teal500": "#02999A",
          "text": "#212121",
          "white": "#FFFFFF",
        },
        "fontColors": Object {
          "accountNavigation": "#FFFFFF",
          "altWhite": "#FFFFFF",
          "autosExpression": "#ffffff",
          "background": "#FAFAFA",
          "black": "#000000",
          "blue400": "#4299E1",
          "checkboxUnselected": "rgba(0, 0, 0, 0.62)",
          "client": "#E0E0E0",
          "createNew": "#212121",
          "darkgray1": "#777777",
          "darkgray2": "#454545",
          "darkgray3": "#424242",
          "dropdown": "#212121",
          "gray1": "#E0E0E0",
          "gray100": "#F0F0F0",
          "gray2": "#E7E7E7",
          "grayAlpha": "rgba(0,0,0,0.17)",
          "grayAlpha100": "rgba(0,0,0,0.06)",
          "grayAlpha400": "rgba(0,0,0,0.36)",
          "green600": "#2F855A",
          "inverseText": "#212121",
          "medgray1": "#757575",
          "medgray2": "#888888",
          "menuDark": "#21262A",
          "menuLight": "#2E3841",
          "orange400": "#ED8936",
          "pink400": "#ED64A6",
          "primary1": "#5548D9",
          "primary2": "#212121",
          "primary3": "#00796B",
          "primary4": "#D32F2F",
          "primary5": "#f0b941",
          "purple400": "#9f7aea",
          "red400": "#f56565",
          "red600": "#C53030",
          "secondary1": "#2E7D32",
          "secondary2": "#757575",
          "secondary3": "#673AB7",
          "secondary4": "#9C27B0",
          "secondary5": "#00796B",
          "secondary6": "#ECF0FC",
          "submenu": "#FFFFFF",
          "teal400": "#38B2AC",
          "teal500": "#02999A",
          "text": "#212121",
          "white": "#FFFFFF",
        },
      }
    }
    style={
      Object {
        "color": "blue",
      }
    }
    type="submit"
  >
    <StyledButton
      boxAs="a"
      buttonSize="md"
      className="myClass"
      disabled={true}
      href="www.moosend.com"
      id="button1"
      onClick={[Function]}
      palette={
        Object {
          "backgroundColors": Object {
            "accountNavigation": "#FFFFFF",
            "altWhite": "#FFFFFF",
            "autosExpression": "#ffffff",
            "background": "#FAFAFA",
            "black": "#000000",
            "blue400": "#4299E1",
            "checkboxUnselected": "rgba(0, 0, 0, 0.62)",
            "client": "#E0E0E0",
            "createNew": "#FFFFFF",
            "darkgray1": "#777777",
            "darkgray2": "#454545",
            "darkgray3": "#424242",
            "dropdown": "#FFFFFF",
            "gray1": "#E0E0E0",
            "gray100": "#F0F0F0",
            "gray2": "#E7E7E7",
            "grayAlpha": "rgba(0,0,0,0.17)",
            "grayAlpha100": "rgba(0,0,0,0.06)",
            "grayAlpha400": "rgba(0,0,0,0.36)",
            "green600": "#2F855A",
            "inverseText": "#212121",
            "medgray1": "#757575",
            "medgray2": "#888888",
            "menuDark": "#21262A",
            "menuLight": "#2E3841",
            "orange400": "#ED8936",
            "pink400": "#ED64A6",
            "primary1": "#5548D9",
            "primary2": "#212121",
            "primary3": "#00796B",
            "primary4": "#D32F2F",
            "primary5": "#f0b941",
            "purple400": "#9f7aea",
            "red400": "#f56565",
            "red600": "#C53030",
            "secondary1": "#2E7D32",
            "secondary2": "#757575",
            "secondary3": "#673AB7",
            "secondary4": "#9C27B0",
            "secondary5": "#00796B",
            "secondary6": "#ECF0FC",
            "submenu": "#FFFFFF",
            "teal400": "#38B2AC",
            "teal500": "#02999A",
            "text": "#212121",
            "white": "#FFFFFF",
          },
          "borderColors": Object {
            "accountNavigation": "#FFFFFF",
            "altWhite": "#FFFFFF",
            "autosExpression": "#ffffff",
            "background": "#FAFAFA",
            "black": "#000000",
            "blue400": "#4299E1",
            "checkboxUnselected": "rgba(0, 0, 0, 0.62)",
            "client": "#E0E0E0",
            "createNew": "#FFFFFF",
            "darkgray1": "#777777",
            "darkgray2": "#454545",
            "darkgray3": "#424242",
            "dropdown": "#FFFFFF",
            "gray1": "#E0E0E0",
            "gray100": "#F0F0F0",
            "gray2": "#E7E7E7",
            "grayAlpha": "rgba(0,0,0,0.17)",
            "grayAlpha100": "rgba(0,0,0,0.06)",
            "grayAlpha400": "rgba(0,0,0,0.36)",
            "green600": "#2F855A",
            "inverseText": "#212121",
            "medgray1": "#757575",
            "medgray2": "#888888",
            "menuDark": "#21262A",
            "menuLight": "#2E3841",
            "orange400": "#ED8936",
            "pink400": "#ED64A6",
            "primary1": "#5548D9",
            "primary2": "#212121",
            "primary3": "#00796B",
            "primary4": "#D32F2F",
            "primary5": "#f0b941",
            "purple400": "#9f7aea",
            "red400": "#f56565",
            "red600": "#C53030",
            "secondary1": "#2E7D32",
            "secondary2": "#757575",
            "secondary3": "#673AB7",
            "secondary4": "#9C27B0",
            "secondary5": "#00796B",
            "secondary6": "#ECF0FC",
            "submenu": "#FFFFFF",
            "teal400": "#38B2AC",
            "teal500": "#02999A",
            "text": "#212121",
            "white": "#FFFFFF",
          },
          "fontColors": Object {
            "accountNavigation": "#FFFFFF",
            "altWhite": "#FFFFFF",
            "autosExpression": "#ffffff",
            "background": "#FAFAFA",
            "black": "#000000",
            "blue400": "#4299E1",
            "checkboxUnselected": "rgba(0, 0, 0, 0.62)",
            "client": "#E0E0E0",
            "createNew": "#212121",
            "darkgray1": "#777777",
            "darkgray2": "#454545",
            "darkgray3": "#424242",
            "dropdown": "#212121",
            "gray1": "#E0E0E0",
            "gray100": "#F0F0F0",
            "gray2": "#E7E7E7",
            "grayAlpha": "rgba(0,0,0,0.17)",
            "grayAlpha100": "rgba(0,0,0,0.06)",
            "grayAlpha400": "rgba(0,0,0,0.36)",
            "green600": "#2F855A",
            "inverseText": "#212121",
            "medgray1": "#757575",
            "medgray2": "#888888",
            "menuDark": "#21262A",
            "menuLight": "#2E3841",
            "orange400": "#ED8936",
            "pink400": "#ED64A6",
            "primary1": "#5548D9",
            "primary2": "#212121",
            "primary3": "#00796B",
            "primary4": "#D32F2F",
            "primary5": "#f0b941",
            "purple400": "#9f7aea",
            "red400": "#f56565",
            "red600": "#C53030",
            "secondary1": "#2E7D32",
            "secondary2": "#757575",
            "secondary3": "#673AB7",
            "secondary4": "#9C27B0",
            "secondary5": "#00796B",
            "secondary6": "#ECF0FC",
            "submenu": "#FFFFFF",
            "teal400": "#38B2AC",
            "teal500": "#02999A",
            "text": "#212121",
            "white": "#FFFFFF",
          },
        }
      }
      style={
        Object {
          "color": "blue",
        }
      }
      type="submit"
    >
      <Box
        boxAs="a"
        buttonSize="md"
        className="sc-gsDKAQ sc-dkPtRN boMExE iXdnUr myClass"
        disabled={true}
        href="www.moosend.com"
        id="button1"
        onClick={[Function]}
        palette={
          Object {
            "backgroundColors": Object {
              "accountNavigation": "#FFFFFF",
              "altWhite": "#FFFFFF",
              "autosExpression": "#ffffff",
              "background": "#FAFAFA",
              "black": "#000000",
              "blue400": "#4299E1",
              "checkboxUnselected": "rgba(0, 0, 0, 0.62)",
              "client": "#E0E0E0",
              "createNew": "#FFFFFF",
              "darkgray1": "#777777",
              "darkgray2": "#454545",
              "darkgray3": "#424242",
              "dropdown": "#FFFFFF",
              "gray1": "#E0E0E0",
              "gray100": "#F0F0F0",
              "gray2": "#E7E7E7",
              "grayAlpha": "rgba(0,0,0,0.17)",
              "grayAlpha100": "rgba(0,0,0,0.06)",
              "grayAlpha400": "rgba(0,0,0,0.36)",
              "green600": "#2F855A",
              "inverseText": "#212121",
              "medgray1": "#757575",
              "medgray2": "#888888",
              "menuDark": "#21262A",
              "menuLight": "#2E3841",
              "orange400": "#ED8936",
              "pink400": "#ED64A6",
              "primary1": "#5548D9",
              "primary2": "#212121",
              "primary3": "#00796B",
              "primary4": "#D32F2F",
              "primary5": "#f0b941",
              "purple400": "#9f7aea",
              "red400": "#f56565",
              "red600": "#C53030",
              "secondary1": "#2E7D32",
              "secondary2": "#757575",
              "secondary3": "#673AB7",
              "secondary4": "#9C27B0",
              "secondary5": "#00796B",
              "secondary6": "#ECF0FC",
              "submenu": "#FFFFFF",
              "teal400": "#38B2AC",
              "teal500": "#02999A",
              "text": "#212121",
              "white": "#FFFFFF",
            },
            "borderColors": Object {
              "accountNavigation": "#FFFFFF",
              "altWhite": "#FFFFFF",
              "autosExpression": "#ffffff",
              "background": "#FAFAFA",
              "black": "#000000",
              "blue400": "#4299E1",
              "checkboxUnselected": "rgba(0, 0, 0, 0.62)",
              "client": "#E0E0E0",
              "createNew": "#FFFFFF",
              "darkgray1": "#777777",
              "darkgray2": "#454545",
              "darkgray3": "#424242",
              "dropdown": "#FFFFFF",
              "gray1": "#E0E0E0",
              "gray100": "#F0F0F0",
              "gray2": "#E7E7E7",
              "grayAlpha": "rgba(0,0,0,0.17)",
              "grayAlpha100": "rgba(0,0,0,0.06)",
              "grayAlpha400": "rgba(0,0,0,0.36)",
              "green600": "#2F855A",
              "inverseText": "#212121",
              "medgray1": "#757575",
              "medgray2": "#888888",
              "menuDark": "#21262A",
              "menuLight": "#2E3841",
              "orange400": "#ED8936",
              "pink400": "#ED64A6",
              "primary1": "#5548D9",
              "primary2": "#212121",
              "primary3": "#00796B",
              "primary4": "#D32F2F",
              "primary5": "#f0b941",
              "purple400": "#9f7aea",
              "red400": "#f56565",
              "red600": "#C53030",
              "secondary1": "#2E7D32",
              "secondary2": "#757575",
              "secondary3": "#673AB7",
              "secondary4": "#9C27B0",
              "secondary5": "#00796B",
              "secondary6": "#ECF0FC",
              "submenu": "#FFFFFF",
              "teal400": "#38B2AC",
              "teal500": "#02999A",
              "text": "#212121",
              "white": "#FFFFFF",
            },
            "fontColors": Object {
              "accountNavigation": "#FFFFFF",
              "altWhite": "#FFFFFF",
              "autosExpression": "#ffffff",
              "background": "#FAFAFA",
              "black": "#000000",
              "blue400": "#4299E1",
              "checkboxUnselected": "rgba(0, 0, 0, 0.62)",
              "client": "#E0E0E0",
              "createNew": "#212121",
              "darkgray1": "#777777",
              "darkgray2": "#454545",
              "darkgray3": "#424242",
              "dropdown": "#212121",
              "gray1": "#E0E0E0",
              "gray100": "#F0F0F0",
              "gray2": "#E7E7E7",
              "grayAlpha": "rgba(0,0,0,0.17)",
              "grayAlpha100": "rgba(0,0,0,0.06)",
              "grayAlpha400": "rgba(0,0,0,0.36)",
              "green600": "#2F855A",
              "inverseText": "#212121",
              "medgray1": "#757575",
              "medgray2": "#888888",
              "menuDark": "#21262A",
              "menuLight": "#2E3841",
              "orange400": "#ED8936",
              "pink400": "#ED64A6",
              "primary1": "#5548D9",
              "primary2": "#212121",
              "primary3": "#00796B",
              "primary4": "#D32F2F",
              "primary5": "#f0b941",
              "purple400": "#9f7aea",
              "red400": "#f56565",
              "red600": "#C53030",
              "secondary1": "#2E7D32",
              "secondary2": "#757575",
              "secondary3": "#673AB7",
              "secondary4": "#9C27B0",
              "secondary5": "#00796B",
              "secondary6": "#ECF0FC",
              "submenu": "#FFFFFF",
              "teal400": "#38B2AC",
              "teal500": "#02999A",
              "text": "#212121",
              "white": "#FFFFFF",
            },
          }
        }
        style={
          Object {
            "color": "blue",
          }
        }
        transition="0.2s all"
        type="submit"
      >
        <Component
          boxAs="a"
          buttonSize="md"
          className="sc-gsDKAQ sc-dkPtRN boMExE iXdnUr myClass"
          disabled={true}
          href="www.moosend.com"
          id="button1"
          onClick={[Function]}
          palette={
            Object {
              "backgroundColors": Object {
                "accountNavigation": "#FFFFFF",
                "altWhite": "#FFFFFF",
                "autosExpression": "#ffffff",
                "background": "#FAFAFA",
                "black": "#000000",
                "blue400": "#4299E1",
                "checkboxUnselected": "rgba(0, 0, 0, 0.62)",
                "client": "#E0E0E0",
                "createNew": "#FFFFFF",
                "darkgray1": "#777777",
                "darkgray2": "#454545",
                "darkgray3": "#424242",
                "dropdown": "#FFFFFF",
                "gray1": "#E0E0E0",
                "gray100": "#F0F0F0",
                "gray2": "#E7E7E7",
                "grayAlpha": "rgba(0,0,0,0.17)",
                "grayAlpha100": "rgba(0,0,0,0.06)",
                "grayAlpha400": "rgba(0,0,0,0.36)",
                "green600": "#2F855A",
                "inverseText": "#212121",
                "medgray1": "#757575",
                "medgray2": "#888888",
                "menuDark": "#21262A",
                "menuLight": "#2E3841",
                "orange400": "#ED8936",
                "pink400": "#ED64A6",
                "primary1": "#5548D9",
                "primary2": "#212121",
                "primary3": "#00796B",
                "primary4": "#D32F2F",
                "primary5": "#f0b941",
                "purple400": "#9f7aea",
                "red400": "#f56565",
                "red600": "#C53030",
                "secondary1": "#2E7D32",
                "secondary2": "#757575",
                "secondary3": "#673AB7",
                "secondary4": "#9C27B0",
                "secondary5": "#00796B",
                "secondary6": "#ECF0FC",
                "submenu": "#FFFFFF",
                "teal400": "#38B2AC",
                "teal500": "#02999A",
                "text": "#212121",
                "white": "#FFFFFF",
              },
              "borderColors": Object {
                "accountNavigation": "#FFFFFF",
                "altWhite": "#FFFFFF",
                "autosExpression": "#ffffff",
                "background": "#FAFAFA",
                "black": "#000000",
                "blue400": "#4299E1",
                "checkboxUnselected": "rgba(0, 0, 0, 0.62)",
                "client": "#E0E0E0",
                "createNew": "#FFFFFF",
                "darkgray1": "#777777",
                "darkgray2": "#454545",
                "darkgray3": "#424242",
                "dropdown": "#FFFFFF",
                "gray1": "#E0E0E0",
                "gray100": "#F0F0F0",
                "gray2": "#E7E7E7",
                "grayAlpha": "rgba(0,0,0,0.17)",
                "grayAlpha100": "rgba(0,0,0,0.06)",
                "grayAlpha400": "rgba(0,0,0,0.36)",
                "green600": "#2F855A",
                "inverseText": "#212121",
                "medgray1": "#757575",
                "medgray2": "#888888",
                "menuDark": "#21262A",
                "menuLight": "#2E3841",
                "orange400": "#ED8936",
                "pink400": "#ED64A6",
                "primary1": "#5548D9",
                "primary2": "#212121",
                "primary3": "#00796B",
                "primary4": "#D32F2F",
                "primary5": "#f0b941",
                "purple400": "#9f7aea",
                "red400": "#f56565",
                "red600": "#C53030",
                "secondary1": "#2E7D32",
                "secondary2": "#757575",
                "secondary3": "#673AB7",
                "secondary4": "#9C27B0",
                "secondary5": "#00796B",
                "secondary6": "#ECF0FC",
                "submenu": "#FFFFFF",
                "teal400": "#38B2AC",
                "teal500": "#02999A",
                "text": "#212121",
                "white": "#FFFFFF",
              },
              "fontColors": Object {
                "accountNavigation": "#FFFFFF",
                "altWhite": "#FFFFFF",
                "autosExpression": "#ffffff",
                "background": "#FAFAFA",
                "black": "#000000",
                "blue400": "#4299E1",
                "checkboxUnselected": "rgba(0, 0, 0, 0.62)",
                "client": "#E0E0E0",
                "createNew": "#212121",
                "darkgray1": "#777777",
                "darkgray2": "#454545",
                "darkgray3": "#424242",
                "dropdown": "#212121",
                "gray1": "#E0E0E0",
                "gray100": "#F0F0F0",
                "gray2": "#E7E7E7",
                "grayAlpha": "rgba(0,0,0,0.17)",
                "grayAlpha100": "rgba(0,0,0,0.06)",
                "grayAlpha400": "rgba(0,0,0,0.36)",
                "green600": "#2F855A",
                "inverseText": "#212121",
                "medgray1": "#757575",
                "medgray2": "#888888",
                "menuDark": "#21262A",
                "menuLight": "#2E3841",
                "orange400": "#ED8936",
                "pink400": "#ED64A6",
                "primary1": "#5548D9",
                "primary2": "#212121",
                "primary3": "#00796B",
                "primary4": "#D32F2F",
                "primary5": "#f0b941",
                "purple400": "#9f7aea",
                "red400": "#f56565",
                "red600": "#C53030",
                "secondary1": "#2E7D32",
                "secondary2": "#757575",
                "secondary3": "#673AB7",
                "secondary4": "#9C27B0",
                "secondary5": "#00796B",
                "secondary6": "#ECF0FC",
                "submenu": "#FFFFFF",
                "teal400": "#38B2AC",
                "teal500": "#02999A",
                "text": "#212121",
                "white": "#FFFFFF",
              },
            }
          }
          style={
            Object {
              "color": "blue",
            }
          }
          transition="0.2s all"
          type="submit"
        >
          <StyledBox
            as="a"
            boxAs="a"
            buttonSize="md"
            className="sc-gsDKAQ sc-dkPtRN boMExE iXdnUr myClass"
            disabled={true}
            href="www.moosend.com"
            id="button1"
            onClick={[Function]}
            palette={
              Object {
                "backgroundColors": Object {
                  "accountNavigation": "#FFFFFF",
                  "altWhite": "#FFFFFF",
                  "autosExpression": "#ffffff",
                  "background": "#FAFAFA",
                  "black": "#000000",
                  "blue400": "#4299E1",
                  "checkboxUnselected": "rgba(0, 0, 0, 0.62)",
                  "client": "#E0E0E0",
                  "createNew": "#FFFFFF",
                  "darkgray1": "#777777",
                  "darkgray2": "#454545",
                  "darkgray3": "#424242",
                  "dropdown": "#FFFFFF",
                  "gray1": "#E0E0E0",
                  "gray100": "#F0F0F0",
                  "gray2": "#E7E7E7",
                  "grayAlpha": "rgba(0,0,0,0.17)",
                  "grayAlpha100": "rgba(0,0,0,0.06)",
                  "grayAlpha400": "rgba(0,0,0,0.36)",
                  "green600": "#2F855A",
                  "inverseText": "#212121",
                  "medgray1": "#757575",
                  "medgray2": "#888888",
                  "menuDark": "#21262A",
                  "menuLight": "#2E3841",
                  "orange400": "#ED8936",
                  "pink400": "#ED64A6",
                  "primary1": "#5548D9",
                  "primary2": "#212121",
                  "primary3": "#00796B",
                  "primary4": "#D32F2F",
                  "primary5": "#f0b941",
                  "purple400": "#9f7aea",
                  "red400": "#f56565",
                  "red600": "#C53030",
                  "secondary1": "#2E7D32",
                  "secondary2": "#757575",
                  "secondary3": "#673AB7",
                  "secondary4": "#9C27B0",
                  "secondary5": "#00796B",
                  "secondary6": "#ECF0FC",
                  "submenu": "#FFFFFF",
                  "teal400": "#38B2AC",
                  "teal500": "#02999A",
                  "text": "#212121",
                  "white": "#FFFFFF",
                },
                "borderColors": Object {
                  "accountNavigation": "#FFFFFF",
                  "altWhite": "#FFFFFF",
                  "autosExpression": "#ffffff",
                  "background": "#FAFAFA",
                  "black": "#000000",
                  "blue400": "#4299E1",
                  "checkboxUnselected": "rgba(0, 0, 0, 0.62)",
                  "client": "#E0E0E0",
                  "createNew": "#FFFFFF",
                  "darkgray1": "#777777",
                  "darkgray2": "#454545",
                  "darkgray3": "#424242",
                  "dropdown": "#FFFFFF",
                  "gray1": "#E0E0E0",
                  "gray100": "#F0F0F0",
                  "gray2": "#E7E7E7",
                  "grayAlpha": "rgba(0,0,0,0.17)",
                  "grayAlpha100": "rgba(0,0,0,0.06)",
                  "grayAlpha400": "rgba(0,0,0,0.36)",
                  "green600": "#2F855A",
                  "inverseText": "#212121",
                  "medgray1": "#757575",
                  "medgray2": "#888888",
                  "menuDark": "#21262A",
                  "menuLight": "#2E3841",
                  "orange400": "#ED8936",
                  "pink400": "#ED64A6",
                  "primary1": "#5548D9",
                  "primary2": "#212121",
                  "primary3": "#00796B",
                  "primary4": "#D32F2F",
                  "primary5": "#f0b941",
                  "purple400": "#9f7aea",
                  "red400": "#f56565",
                  "red600": "#C53030",
                  "secondary1": "#2E7D32",
                  "secondary2": "#757575",
                  "secondary3": "#673AB7",
                  "secondary4": "#9C27B0",
                  "secondary5": "#00796B",
                  "secondary6": "#ECF0FC",
                  "submenu": "#FFFFFF",
                  "teal400": "#38B2AC",
                  "teal500": "#02999A",
                  "text": "#212121",
                  "white": "#FFFFFF",
                },
                "fontColors": Object {
                  "accountNavigation": "#FFFFFF",
                  "altWhite": "#FFFFFF",
                  "autosExpression": "#ffffff",
                  "background": "#FAFAFA",
                  "black": "#000000",
                  "blue400": "#4299E1",
                  "checkboxUnselected": "rgba(0, 0, 0, 0.62)",
                  "client": "#E0E0E0",
                  "createNew": "#212121",
                  "darkgray1": "#777777",
                  "darkgray2": "#454545",
                  "darkgray3": "#424242",
                  "dropdown": "#212121",
                  "gray1": "#E0E0E0",
                  "gray100": "#F0F0F0",
                  "gray2": "#E7E7E7",
                  "grayAlpha": "rgba(0,0,0,0.17)",
                  "grayAlpha100": "rgba(0,0,0,0.06)",
                  "grayAlpha400": "rgba(0,0,0,0.36)",
                  "green600": "#2F855A",
                  "inverseText": "#212121",
                  "medgray1": "#757575",
                  "medgray2": "#888888",
                  "menuDark": "#21262A",
                  "menuLight": "#2E3841",
                  "orange400": "#ED8936",
                  "pink400": "#ED64A6",
                  "primary1": "#5548D9",
                  "primary2": "#212121",
                  "primary3": "#00796B",
                  "primary4": "#D32F2F",
                  "primary5": "#f0b941",
                  "purple400": "#9f7aea",
                  "red400": "#f56565",
                  "red600": "#C53030",
                  "secondary1": "#2E7D32",
                  "secondary2": "#757575",
                  "secondary3": "#673AB7",
                  "secondary4": "#9C27B0",
                  "secondary5": "#00796B",
                  "secondary6": "#ECF0FC",
                  "submenu": "#FFFFFF",
                  "teal400": "#38B2AC",
                  "teal500": "#02999A",
                  "text": "#212121",
                  "white": "#FFFFFF",
                },
              }
            }
            style={
              Object {
                "color": "blue",
              }
            }
            transition="0.2s all"
            type="submit"
          >
            <a
              className="sc-bdvvtL buKeIX sc-gsDKAQ sc-dkPtRN boMExE iXdnUr myClass"
              disabled={true}
              href="www.moosend.com"
              id="button1"
              onClick={[Function]}
              style={
                Object {
                  "color": "blue",
                }
              }
              type="submit"
            >
              Mooskin
            </a>
          </StyledBox>
        </Component>
      </Box>
    </StyledButton>
  </Component>
</Button>
`;
